src/contracts/core/StrategyManager.sol

```diff
@@ -186,129 +183,38 @@ contract StrategyManager is
         shares = _depositIntoStrategy(staker, strategy, token, amount);
     }
 
-    /**
-     * @notice Called by the DelegationManager as part of the forced undelegation of the @param staker from their delegated operator.
-     * This function queues a withdrawal of all of the `staker`'s shares in EigenLayer to the staker themself, and then undelegates the staker.
-     * The staker will consequently be able to complete this withdrawal by calling the `completeQueuedWithdrawal` function.
-     * @param staker The staker to force-undelegate.
-     * @dev Returns: an array of strategies withdrawn from, the shares withdrawn from each strategy, and the root of the newly queued withdrawal.
-     */
-    function forceTotalWithdrawal(
-        address staker
-    )
-        external
-        onlyDelegationManager
-        onlyWhenNotPaused(PAUSED_WITHDRAWALS)
-        onlyNotFrozen(staker)
-        nonReentrant
-        returns (IStrategy[] memory, uint256[] memory, bytes32)
-    {
-        uint256 strategiesLength = stakerStrategyList[staker].length;
-        IStrategy[] memory strategies = new IStrategy[](strategiesLength);
-        uint256[] memory shares = new uint256[](strategiesLength);
-        uint256[] memory strategyIndexes = new uint256[](strategiesLength);
-
-        for (uint256 i = 0; i < strategiesLength; ) {
-            uint256 index = (strategiesLength - 1) - i;
-            strategies[i] = stakerStrategyList[staker][index];
-            shares[i] = stakerStrategyShares[staker][strategies[i]];
-            strategyIndexes[i] = index;
-            unchecked {
-                ++i;
-            }
-        }
-        bytes32 queuedWithdrawal = _queueWithdrawal(staker, strategyIndexes, strategies, shares, staker);
-        return (strategies, shares, queuedWithdrawal);
+    function removeShares(
```

Comment:
natspec comments needed in this file in general?

Reply:
I've added a bunch. please note specific places and/or just add more comments yourself